chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -591,10 +591,34 @@ where
         tx_source: TransactionSource,
         tx: &Transaction,
     ) -> Result<(), ConnectTransactionError> {
-        // decrement nonce if disconnected input spent from account
+        // Note: it's not strictly required to check inputs/outputs to undo accounting operations
+        //       because `accounting_block_undo` is not empty in that case. But we do it anyway
+        //       to double check that the logic is consistent.
+        let mut apply_accounting_delta_undo = false;
```

Comment:
This mutability is really a bad idea... can't we just do the undo anyway and if it's empty just "do nothing" because it's empty?

Reply:
Ok I will remove these checks and just apply undo deltas if they are available.